File: eCinema/eCinema.UI/ecinema_desktop/lib/models/showtime.dart
import 'package:json_annotation/json_annotation.dart';
import 'movie.dart';
import 'cinema_hall.dart';

part 'showtime.g.dart';

@JsonSerializable()
class Showtime {
  int? id;
  int? movieId;
  int? cinemaHallId;
  DateTime? startTime;
  DateTime? endTime;
  double? basePrice;

  Movie? movie;
  CinemaHall? cinemaHall;

  Showtime({
    this.id,
    this.movieId,
    this.cinemaHallId,
    this.startTime,
    this.endTime,
    this.basePrice,
    this.movie,
    this.cinemaHall,
  });

  factory Showtime.fromJson(Map<String, dynamic> json) =>
      _$ShowtimeFromJson(json);
  Map<String, dynamic> toJson() => _$ShowtimeToJson(this);
}

File: eCinema/eCinema.Models/Entities/Showtime.cs
using eCinema.Model.Entities;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace eCinema.Models.Entities
{
    public class Showtime
    {
        public int Id { get; set; }
        public int MovieId { get; set; }
        public int CinemaHallId { get; set; }
        public DateTime StartTime { get; set; }
        public DateTime EndTime { get; set; }
        public decimal BasePrice { get; set; }

        public Movie Movie { get; set; }
        public CinemaHall CinemaHall { get; set; }
        public ICollection<Booking> Bookings { get; set; } = new List<Booking>();
    }
}
